### DISCUSSION TITLE
How to implement my own STDP rule?

### DISCUSSION THREAD

**Discussion by khollmes** [BindsNET/bindsnet#644] [Q&A] [answered]
Hello, 

I'm a beginner and I'm looking to simulate a SNN based on a memristor, for my grad internship. And I was wondering if I could use Bindsnet library to simulate it. This is pretty new to me and I was wondering if someone could help me getting started.

The function that I'm looking to simulate is pretty much an asymmetric STDP: if delta_t <0  A_pre* exp( -delta_t/tau_pre). And if delta_t >0  A_post* exp( -delta_t/tau_post)

Here we have: delta_t the time difference between pre-synaptic and post-synaptic spikes and delta_w the weight change for the synapse. The constants tau_pre and tau_post are my times constants and A_post and A_pre are I think my learning rate. Those constant are issued from a fit of some data I have.

I've been trying to implement this STDP on my own by adding two traces xpost and xpre to my nodes, and changing the postpre rule to take into account those new traces rather than x, but without much success. I feel like there is a simple and elegant way to do it but I just can't seem to find it.

Maybe I should try to change MSTD?

Thank you in advance for your help.

**Comment by Hananel-Hazan** [comment, marked as answer]
Take a look of the function [`PostPre`](https://github.com/BindsNET/bindsnet/blob/9a2cc55fb223cd05efdaf35eacee655a2c8c9c5d/bindsnet/learning/learning.py#L146). It very close to what you need, you can modify it to fit your desired function.

**Reply by khollmes** [reply]
Thank you for your answer I'll look into it.
